lib/foods/model/food_list_screen.dart

```diff
@@ -0,0 +1,556 @@
+                                  //trueだったらfalse返ってくる(逆も)
+                                  notifier.toggleTagChip(
+                                      _choiceList[index], newBoolValue);
+                                },
+                              ),
+                            );
+                          }),
+                        ),
+                      ],
+                    ),
+                  ),
+                ),
+              ),
+              actions: <Widget>[
+                ElevatedButton(
+                  style: ElevatedButton.styleFrom(
+                    primary: Colors.black26,
+                    side: BorderSide(
+                      color: Colors.black26,
+                      width: 1, //太さ
+                    ),
+                  ),
+                  onPressed: () {
+                    Navigator.pop(context, false);
+                  },
+                  child: Text("キャンセル"),
+                ),
+                ElevatedButton(
+                  style: ElevatedButton.styleFrom(
+                    primary: Colors.black26,
+                    side: BorderSide(
+                      color: Colors.black26,
+                      width: 1, //太さ
+                    ),
+                  ),
+                  onPressed: () async {
+                    //バリデーションない時だけ保存処理実行
+                    if (formKey.currentState!.validate()) {
+                      //tagsの中に、HashMapの中でtrueになったタグのStringが入ってくる
+                      final tags =
+                          getEnableTags(state.tagSelectState ?? HashMap());
+                      //「newFood」が作られる(タイトル、タグ1~5(空のタグ含む) のセット)
+                      final newFood = createSaveData(title.text, tags);
+                      //作った「_newTodo」を_notifierのinsertTodoDataに渡してる
+                      //これで内部的にrepositoryを呼んでDBへの書き込みがされる
+                      notifier.insertFoodData(newFood);
+                      //showDialogでtrueを返す
+                      Navigator.pop(context, true);
+                    }
+                  },
+                  child: Text("保存"),
+                )
+              ],
+            );
+          },
+        );
+      },
+    );
+    if (needRefresh == true) {
+      notifier.getFoodData();
+    }
+  }
+
+  final List<String> _choiceList = [
+    "和食",
+    "洋食",
+    "アジア",
+    "イタリアン",
+    "中華",
+    "海鮮",
+    "朝食",
+    "昼食",
+    "夕食",
+    "鍋もの",
+    "丼もの",
+    "メイン",
+    "サブ",
+    "汁物",
+    "サラダ",
+    "鶏肉",
+    "豚肉",
+    "牛肉",
+    "つまみ",
+    "お菓子",
+    "飲み物",
+    "その他",
+  ];
+
+// HashMapに入ってるValue (タグのON,OFFの状態)のtrueのkeyを抽出して,
+// enableTagsというListを作っている
+// Zennメモあり
+  List<String> getEnableTags(HashMap<String, bool> tags) {
+    List<String> enableTags = [];
+    // ここで絞り込みを行う
+    // HashMap. keyのListを取得できる。このリストを使って、ループを行って、フィルターを使う
+    tags.entries.where((element) => element.value).forEach((element) {
+      enableTags.add(element.key);
+    });
+    return enableTags;
+  }
```

Comment:
```suggestion
    final enableTags = tags.entries.where((element) => element.key).toList();
```

間違っていたら申し訳ないのですが、上記の感じでも同じようにデータ取れないでしょうか？
動作確認していないので確認お願いします！

Reply:
この修正はしない（Slackのやりとりにて）